src/main/java/com/microsoft/sqlserver/jdbc/SQLServerClob.java

```diff
@@ -91,7 +88,9 @@ final JDBCType getJdbcType() {
     private ArrayList<Closeable> activeStreams = new ArrayList<>(1);
 
     transient SQLServerConnection con;
-    private static Logger logger;
+    
+    private final Logger logger = Logger.getLogger(getClass().getName());
```

Comment:
There is now a logger lookup whenever an instance of `SQLServerClob` or `SQLServerNClob` is created. This could be avoided if:
1. there is a `static final` logger in `SQLServerClob` or `SQLServerNClob`, like before
1. the logger in `SQLServerClobBase` is made `private final`, like in this PR
1. `SQLServerClob` and `SQLServerNClob` pass the logger in the `super` constructor call

Reply:
Hi @marschall 

Thanks for replying back! We could go for either solution, but I removed the loggers from sub classes as: 
1. Class Lookup isn't expensive here.
2. The loggers in sub classes are not used anywhere else except being passed in constructor (but remain attached in memory with class objects)

Reply:
It's not the class lookup that I was concerned about, it's get logger lookup.

Reply:
Can you please explain what you mean by Logger lookup and how will it be different in the other case - maybe I am not able to understand your point.

Reply:
With a non-static logger in `SQLServerClob` or `SQLServerNClob` every time a new instance of these classes is generated the following is executed:
```java
Logger.getLogger(getClass().getName())
```
If we ignore `getClass().getName()` for now that still leaves us with `Logger.getLogger()` which does among other things:
1. `Reflection.getCallerClass()` which creates an exception and walks the entire stack
1. check for a security manager several times
1. goes through a global lock in `java.util.logging.LogManager.LoggerContext#namedLoggers`

If you make the loggers in `SQLServerClob` and `SQLServerNClob` `static` all that work happens only one during class loading.

Reply:
Hi @marschall I understand the concern for non-static logger here. Made changes in the new PR #563 as discussed above. Request you to review the PR so we can get it working.